segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/SegmentMetadataUpdateTransaction.java

```diff
@@ -455,6 +461,28 @@ void preProcessAsSourceSegment(MergeSegmentOperation operation) throws StreamSeg
         }
     }
 
+    /**
+     * Returns the total number of unique ExtendedAttributes that would exist for the segment assuming the
+     * {@link AttributeUpdateCollection} would be applied.
+     *
+     * @param attributeUpdates The {@link AttributeUpdateCollection} to apply.
+     * @return The number of unique extended attributes.
+     */
+    private int getUniqueExtendedAttributesCount(AttributeUpdateCollection attributeUpdates) {
+        int count = 0;
+        for (Map.Entry<AttributeId, Long> attribute : baseAttributeValues.entrySet()) {
+            count += !Attributes.isCoreAttribute(attribute.getKey()) ? 1 : 0;
+        }
+        Collection<AttributeId> unique = new HashSet<>();
+        for (AttributeUpdate attribute : attributeUpdates) {
```

Comment:
AttributeUpdates are not duplicated. There's no reason to check for uniqueness. I think we can get rid of the `unique` variable.